src/main/java/com/techiteasy/TechItEasy/security/JwtRequestFilter.java

```diff
@@ -0,0 +1,57 @@
+package com.techiteasy.TechItEasy.security;
+
+import com.techiteasy.TechItEasy.services.CustomUserDetailService;
+import com.techiteasy.TechItEasy.utils.JwtUtil;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+public class JwtRequestFilter extends OncePerRequestFilter {
+
+    private final CustomUserDetailService customUserDetailService;
+    private final JwtUtil jwtUtil;
+
+    public JwtRequestFilter(CustomUserDetailService customUserDetailService, JwtUtil jwtUtil) {
+        this.customUserDetailService = customUserDetailService;
+        this.jwtUtil = jwtUtil;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+
+        final String authorizationHeader = request.getHeader("Authorization");
+
```

Comment:
Als je hier bijvoorbeeld dit neerzet:

  ```
final String apiKey = request.getHeader("ApiKey");

System.out.println(apiKey);
```
        
      
Zou je nog een check kunnen doen of de api key overeen komt en of de gebruiker dus de api mag gebruiken